### PR TITLE
[JUJU-1482] Moved case BOOTSTRAP_PROVIDER logic from task.sh

### PR DESCRIPTION
**Pull request by anvial**
Generating constraints tests in juju-qa-jenking caused naming issues. The possible reason is that 'case BOOTSTRAP_PROVIDER' logic are task.sh file. This patch moves 'case BOOTSTRAP_PROVIDER' to constraints.sh.

## Checklist
 - [ ] Comments answer the question of why design decisions were made

## QA steps

```sh
cd tests
./main -v -p lxd constraints
./main -v -p aws constraints
```

## Documentation changes

None

## Bug reference

None

### CODE REVIEW

File: tests/suites/constraints/constraints.sh [hunk 1]
```diff
@@ -1,88 +1,25 @@
...
-  		set_verbosity
   		cd .. || exit
-  		run "run_constraints_aws"
+
+  		case "${BOOTSTRAP_PROVIDER:-}" in
+      "lxd" | "lxd-remote" | "localhost")
+        run "run_constraints_lxd"
+        ;;
+      "aws" | "ec2")
+        export BOOTSTRAP_PROVIDER="manual"
```
Comment (jack-w-shaw) [inline tests/suites/constraints/constraints.sh:17]: Why is this here?

Comment (wallyworld) [inline tests/suites/constraints/constraints.sh]: why set bootstrap provider to manual?

File: tests/suites/constraints/constraints.sh [hunk 2]
```diff
@@ -1,88 +1,25 @@
...
-  		return
-  fi
 
-  (
-  		set_verbosity
   		cd .. || exit
-  		run "run_constraints_aws"
+
+  		case "${BOOTSTRAP_PROVIDER:-}" in
+      "lxd" | "lxd-remote" | "localhost")
```
Comment (wallyworld) [inline tests/suites/constraints/constraints.sh:13]: indentation looks wrong

File: tests/suites/constraints/constraints.sh (hunk 1 above)
Comment (anvial) [reply tests/suites/constraints/constraints.sh]: fixed

File: tests/suites/constraints/constraints_aws.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
+run_constraints_aws() {
```
Comment (wallyworld) [inline tests/suites/constraints/constraints_aws.sh:1]: we don't need an aws test specially
we just want to distinguish between a container scenario vs a vm scenario

we just want to have run_constraints() and run_constraints_lxd()
the case bootstrap_provider would call the lxd verson for lxd and the "normal" one for everything else

File: tests/suites/constraints/constraints_aws.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  echo
+
+  file="${TEST_DIR}/constraints-aws.txt"
+
+  ensure "constraints-aws" "${file}"
+
+  echo "Deploy 3 machines with different constraints"
+  juju add-machine --constraints "root-disk=16G"
+  juju add-machine --constraints "cores=4 root-disk=16G"
+  juju add-machine --constraints "instance-type=t2.nano"
```
Comment (wallyworld) [inline tests/suites/constraints/constraints_aws.sh:12]: this is aws specific so we can just drop it

**Review by wallyworld — COMMENTED** [review]
See comments in line.
The selection of test based on provider type is not quite right

File: tests/suites/constraints/constraints_aws.sh (hunk 3 above)
Comment (jack-w-shaw) [reply tests/suites/constraints/constraints_aws.sh:1]: Are you sure? Many clouds support different constraints, all with different params and the like. I guess we could test just the lowest common denominator though

File: tests/suites/constraints/constraints.sh (hunk 1 above)
Comment (anvial) [reply tests/suites/constraints/constraints.sh]: This is copy-paste issues, that I didn't catch on local tests (because test only on lxd). Already fixed

File: tests/suites/constraints/constraints_aws.sh (hunk 3 above)
Comment (anvial) [reply tests/suites/constraints/constraints_aws.sh:1]: I think that there will be problems with 'instance-type' constraint check, because it is valid only for AWS.

So if we wish to make 'LXD' and 'non-LXD' tests, we need to drop  'instance-type' constraint test.

File: tests/suites/constraints/constraints_aws.sh (hunk 4 above)
Comment (anvial) [reply tests/suites/constraints/constraints_aws.sh:12]: Ok, are we sure that we wish to drop aws specific test? 

If so, I will rename 'lxd' --> 'containrer', and 'aws' --> 'vm'. 
And further will allow 'vm' on all non-lxd and non-k8s substrates.

**Review by wallyworld — APPROVED** [review]
